### ISSUE TITLE
Is it possible to specify a starting function

### ISSUE THREAD

**Issue by jherbel** [trevorstephens/gplearn#36]
For a number of our problems, we have a rough of a functional form that should be close to the "right answer". Is there a way to feed this in as the initial starting point for the symbolic regression algorithm?

**Comment by trevorstephens** [comment]
No, this is not currently implemented. Will leave the issue open to determine if there's much interest in this.

**Comment by jherbel** [comment]
Ok, thanks!

**Comment by SaremS** [comment]
Would be very helpful if this was implemented.

**Comment by pchankh** [comment]
Yes, I second this feature.

**Comment by AnaKolody** [comment]
I would also be interested in this feature.

**Comment by eggachecat** [comment]
Current you can use some tricks like #39

**Comment by eggachecat** [comment]
#17 related

**Comment by SibylLi** [comment]
Yes it would be really helpful.
btw, what can I do if I really need to specify a staring expression now?

**Comment by trevorstephens** [comment]
If you really, really wanted to, you could run a single generation with the parameters you want to use, then replace all the individual `Program` objects with whatever you wanted to "seed" the population with, then continue evolution with `warm_start`. Check out the test suite for some undocumented examples on how to create `Program` objects from scratch, and the docs for warm starting: https://gplearn.readthedocs.io/en/stable/advanced.html#continuing-evolution

The `Program` API is not documented, so you're kind of on your own here if you wish to go down that path.

**Comment by trevorstephens** [comment]
@SibylLi If you really, really wanted to, you could run a single generation with the parameters you want to use, then replace all the individual `Program` objects with whatever you wanted to "seed" the population with, then continue evolution with `warm_start`. Check out the test suite for some undocumented examples on how to create `Program` objects from scratch, and the docs for warm starting: https://gplearn.readthedocs.io/en/stable/advanced.html#continuing-evolution

The `Program` API is not documented, so you're kind of on your own here if you wish to go down that path.

**Comment by Arya0815** [comment]
It would be very helpful!
BTW, could I generate the initial population by considering some prior-knowledge, e.g. the some feature of xdata corresponding to  ydata, instead of randomly generating it? @trevorstephens

**Comment by LaGvidilo** [comment]
"Is it possible to specify a starting function", yes @trevorstephens say how to with the seed param... but every version determine the result... so i mean is it possible and practical to train a model of ML to determine the seed produce the desired first element code... or can just use the "datelining" concept at start of program. and then, never mind the solution choose, use a CRC func to verify the precise version digital print and store in a temporary locate(to never run a 2nd start of verification of determinist specific function desired).

**Comment by trevorstephens** [comment]
I think I will create a notebook to show how this could be "hacked" into `gplearn` which will satisfy this request with no new code to maintain.

**Comment by acylam** [comment]
@trevorstephens Checking if this feature is added or if there's a notebook showing the "hack"?

> I think I will create a notebook to show how this could be "hacked" into `gplearn` which will satisfy this request with no new code to maintain.

**Comment by gbarros92** [comment]
@trevorstephens Checking if this feature is added or if there's a notebook showing the "hack"?

> I think I will create a notebook to show how this could be "hacked" into `gplearn` which will satisfy this request with no new code to maintain.

**Comment by Jatayu-u** [comment]
@trevorstephens Is this implemented? I am in a similar situation where I need to implement this with some starting functions. It would be great if anybody could help.
